internal/server/add_gcm_issue.go

```diff
@@ -0,0 +1,110 @@
+package server
+
+import (
+	"fmt"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/model"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/sirius"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/util"
+	"golang.org/x/sync/errgroup"
+	"net/http"
+)
+
+type AddGcmIssue interface {
+	GetGCMIssueTypes(ctx sirius.Context) ([]model.RefData, error)
+	GetDeputyClient(ctx sirius.Context, caseRecNumber string, deputyId int) (sirius.DeputyClient, error)
+	AddGcmIssue(ctx sirius.Context, caseRecNumber, notes string, gcmIssueType string, deputyId int) error
+}
+
+type AddGcmIssueVars struct {
+	AppVars
+	GcmIssueTypes []model.RefData
+	CaseRecNumber string
+	Client        sirius.DeputyClient
+	GcmIssueType  string
+	Notes         string
+}
+
+func renderTemplateForAddGcmIssue(client AddGcmIssue, tmpl Template) Handler {
+	return func(app AppVars, w http.ResponseWriter, r *http.Request) error {
+
+		app.PageName = "Add a GCM issue"
+		vars := AddGcmIssueVars{
+			AppVars: app,
+		}
+
+		ctx := getContext(r)
+		group, groupCtx := errgroup.WithContext(ctx.Context)
+
+		group.Go(func() error {
+			gcmIssueTypes, err := client.GetGCMIssueTypes(ctx.With(groupCtx))
+			if err != nil {
+				return err
+			}
+			vars.GcmIssueTypes = gcmIssueTypes
+			return nil
+		})
+
+		if err := group.Wait(); err != nil {
+			return err
+		}
+
+		switch r.Method {
+		case http.MethodGet:
+			return tmpl.ExecuteTemplate(w, "page", vars)
+
+		case http.MethodPost:
+			var caseRecNumber = r.PostFormValue("case-number")
+			var gcmIssueType = r.PostFormValue("issue-type")
+			var notes = r.PostFormValue("notes")
+			var searchForClient = r.PostFormValue("search-for-client")
+			var submitForm = r.PostFormValue("submit-form")
```

Comment:
Because the form allows two different forms of submit (one to search the client and the other to submit / add the issue) it was useful for validation/ logic to determine which one had been clicked.